sdk/textanalytics/Azure.AI.TextAnalytics/tests/Infrastructure/TextAnalyticsSampleBase.cs

```diff
@@ -3,17 +3,38 @@
 
 using Azure.AI.TextAnalytics.Tests;
 using Azure.Core.TestFramework;
+using NUnit.Framework;
 
 namespace Azure.AI.TextAnalytics.Samples
 {
     public class TextAnalyticsSampleBase : SamplesBase<TextAnalyticsTestEnvironment>
     {
-        public TextAnalyticsClientOptions CreateSampleOptions()
+        private const int MaxRetriesCount = 12;
+
+        public TextAnalyticsClientOptions CreateSampleOptions(bool useStaticResource = default)
         {
-            var options = new TextAnalyticsClientOptions();
-            options.Retry.MaxRetries = TextAnalyticsClientLiveTestBase.MaxRetriesCount;
+            TextAnalyticsClientOptions options = new()
+            {
+                Audience = TestEnvironment.GetAudience()
+            };
+
+            // While we use a persistent resource for live tests, we need to increase our retries.
+            // We should remove when having dynamic resource again
+            // Issue: https://github.com/Azure/azure-sdk-for-net/issues/25041
+            if (useStaticResource)
+            {
+                options.Retry.MaxRetries = MaxRetriesCount;
+            }
 
             return options;
         }
+
+        internal void IgnoreIfNotPublicCloud()
+        {
+            if (TestEnvironment.GetAudience() != TextAnalyticsAudience.AzurePublicCloud)
+            {
+                Assert.Ignore("Currently, these tests can only be run in the public cloud.");
+            }
+        }
```

Comment:
Why not just add this in the `TextAnalyticsBase` from which every test fixture should derive, IIRC?

Reply:
Semantically, it made more sense to me in the test bases, but this does let us avoid the code duplication, so I moved it.